spec/lib/booker/errors_spec.rb

```diff
@@ -69,6 +69,42 @@
   end
 end
 
+describe Booker::MidPaginationError do
+  describe 'attributes' do
+    it { expect(subject).to respond_to :error_occurred_during_params }
+    it { expect(subject).to respond_to :results_fetched_prior_to_error }
+    it { expect(subject).to respond_to :message }
+    it { expect(subject).to respond_to :error_occurred_during_params= }
+    it { expect(subject).to respond_to :results_fetched_prior_to_error= }
+    it { expect(subject).to respond_to :message= }
+  end
+
+  describe '.new' do
+    let(:error) { Booker::MidPaginationError.new(message, error_occurred_during_params, results_fetched_prior_to_error) }
+
+    before do
+      expect(error.error_occurred_during_params).to eq error_occurred_during_params
+      expect(error.results_fetched_prior_to_error).to eq results_fetched_prior_to_error
+      expect(error.message).to eq message
+    end
+
+    context 'when all atRtrs defined' do
+      let(:error_occurred_during_params) { {PageSize: 1} }
+      let(:results_fetched_prior_to_error) { ['prior results'] }
+      let(:message) { "Special message" }
+
+      it 'sets error data' do; end
```

Comment:
Probably should have a test here.

Reply:
they're defined/called in the before block; I'm testing that all attributes are set both when they are defined and undefined